metabox/metabox/core/runner.py

```diff
@@ -74,158 +75,217 @@ def _formatter(self, record):
                 scn.service_machine.start_user_session()
             elif scn.mode == "local":
                 scn.local_machine = self._load("local", scn.releases[0])
-                scn.local_machine.rollback_to('provisioned')
+                scn.local_machine.rollback_to("provisioned")
                 if scn.launcher:
                     scn.local_machine.put(scn.LAUNCHER_PATH, scn.launcher)
                 scn.local_machine.start_user_session()
-            logger.info("Starting scenario: {}".format(scn.name))
+
+            scenario_description = self._get_scenario_description(scn)
+            logger.info(
+                "Starting scenario ({}/{}): {}".format(idx, total, scenario_description)
+            )
             scn.run()
             if not scn.has_passed():
                 self.failed = True
-                logger.error("[{}][{}] {} scenario has failed.".format(
-                    scn.mode, scn.releases, scn.name))
+                logger.error(scenario_description + " scenario has failed.")
                 if self.hold_on_fail:
                     if scn.mode == "remote":
                         msg = (
                             "You may hop onto the target machines by issuing "
                             "the following commands:\n{}\n{}\n"
-                            "Press enter to continue testing").format(
-                                scn.remote_machine.get_connecting_cmd(),
-                                scn.service_machine.get_connecting_cmd())
+                            "Press enter to continue testing"
+                        ).format(
+                            scn.remote_machine.get_connecting_cmd(),
+                            scn.service_machine.get_connecting_cmd(),
+                        )
                     elif scn.mode == "local":
                         msg = (
                             "You may hop onto the target machine by issuing "
                             "the following command:\n{}\n"
-                            "Press enter to continue testing").format(
-                                scn.local_machine.get_connecting_cmd())
+                            "Press enter to continue testing"
+                        ).format(scn.local_machine.get_connecting_cmd())
                     print(msg)
                     input()
             else:
-                logger.success("[{}][{}] {} scenario has passed.".format(
-                    scn.mode, scn.releases, scn.name))
+                logger.success(scenario_description + " scenario has passed.")
             self.machine_provider.cleanup()
         del self.machine_provider
         stopTime = time.perf_counter()
         timeTaken = stopTime - startTime
-        print('-' * 80)
-        total = len(self.scn_variants)
+        print("-" * 80)
         status = "Ran {} scenario{} in {:.3f}s".format(
-            total, total != 1 and "s" or "", timeTaken)
+            total, total != 1 and "s" or "", timeTaken
```

Comment:
This is hacky and difficult to read.
```
form = "scenario" if total == 1 else "scenarios"
status = "Ran {} {} in {:.0f} .format(total, form)
```
Do we _really_ need a millisecond precision for something that takes long minutes?

Reply:
yes it is and no we don't, I will change it (not a new change of this PR though)

Reply:
Ah, sorry. Yeah, disregard the comment about precision.